pennylane/templates/subroutines/grover.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the Grover Operation template.
+"""
+import pennylane as qml
+from pennylane.operation import AnyWires, Operation
+from pennylane.ops import Hadamard, PauliZ, MultiControlledX
+
+
+class GroverOperator(Operation):
+    r"""Performs the Grover Diffusion Operator.
+
+    .. math::
+
+        G = 2 |s \rangle \langle s | - I
+        = H^{\bigotimes n} \left( 2 |0\rangle \langle 0| - I \right) H^{\bigotimes n}
+
+
+    For this template, the operator is implemented with a layer of Hadamards, an
+    effective multi-controlled Z gate, and another layer of Hadamards.
+
+    .. figure:: ../../_static/templates/subroutines/grover.svg
+        :align: center
+        :width: 60%
+        :target: javascript:void(0);
+
+    Args:
+        wires (Union[Wires, Sequence[int], or int]): the wires to apply to
+        work_wires (Union[Wires, Sequence[int], or int]): these auxiliary wire assist
+            in the decomposition of :class:`~.ops.qubit.MultiControlledX`.
+
+    **Example**
+
+    For this example, we will be using three wires and ``"default.qubit"``:
+
+    .. code-block:: python
+
+        n_wires = 3
+        wires = list(range(n_wires))
+        dev = qml.device('default.qubit', wires=wires)
+
+    The Grover Diffusion Operator amplifies the magnitude of the component with
+    a negative phase.  For example, if we wanted to select out the :math:`|111\rangle`
+    state, we could use an oracle implementing a `CCZ` gate:
```

Comment:
```suggestion
    a negative phase.  For example, if the solution to the search problem is the :math:`|111\rangle`
    state, we require an oracle that flips its phase; this could be implemented using a `CCZ` gate:
```